crates/bevy_ecs/src/reflect.rs

```diff
@@ -27,7 +27,36 @@ pub struct ReflectComponent {
     copy: fn(&World, &mut World, Entity, Entity),
 }
 
+/// The raw functions needed to make up a [`ReflectComponent`].
+/// 
+/// This is used when creating custom implementations of [`ReflectComponent`] with
```

Comment:
We should link to / explain the simpler way here. Ditto for the resource equivalent.